Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/main/home/HomeFragment.kt

```diff
@@ -88,37 +77,43 @@ class HomeFragment : Fragment(R.layout.home_fragment_layout), AutoInject {
             doNavigate(HomeFragmentDirections.actionMainFragmentToSettingsTracingFragment())
         }
 
-        binding.mainTracingAppBarLayout.addOnOffsetChangedListener(
-            AppBarLayout.OnOffsetChangedListener { appBarLayout, verticalOffset ->
-                try {
-                    if (abs(verticalOffset) >= appBarLayout.totalScrollRange) {
-                        binding.toolbar.setBackgroundResource(R.drawable.top_app_bar_shape)
-                    } else {
-                        binding.toolbar.setBackgroundResource(R.color.colorTopBarBackground)
-                    }
-                } catch (e: Exception) {
-                    Timber.e(e, "Listener has been triggered after onDestroyView()")
+        binding.mainTracingAppBarLayout.addOnOffsetChangedListener { appBarLayout, verticalOffset ->
+            try {
+                if (abs(verticalOffset) >= appBarLayout.totalScrollRange) {
+                    binding.toolbar.setBackgroundResource(R.drawable.top_app_bar_shape)
+                } else {
+                    binding.toolbar.setBackgroundResource(R.color.colorTopBarBackground)
                 }
+            } catch (e: Exception) {
+                Timber.e(e, "Listener has been triggered after onDestroyView()")
             }
-        )
+        }
 
         viewModel.showPopUps()
         viewModel.events.observe2(this) { event -> navigate(event) }
         viewModel.homeItems.observe2(this) { homeAdapter.update(it) }
-        viewModel.errorEvent.observe2(this) { it.toErrorDialogBuilder(requireContext()).show() }
+        viewModel.errorEvent.observe2(this) { displayDialog { it.toErrorDialogBuilder(context) } }
```

Comment:
This is still problematic, I'm not sure if the refactoring is not needed , you yourself the designer of the API keeps forgetting it, you can imagine others who don't know how this works actually.

Reply:
I didn't keep forgetting it. It's not like I've done it again, this change was made with those other two in the beginning. I still wouldn't refactor it as this argument could be said about every lamda function in this app.